opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp

```diff
@@ -71,22 +71,19 @@ namespace Opm {
                        const GridProperties<int>* intGridProperties,
                        const GridProperties<double>* doubleGridProperties)
         {
-            const auto iter = std::find_if( values.begin(), values.end(), []( double value ) { return std::isnan(value); });
-            if (iter != values.end()) {
-                const auto& poro = doubleGridProperties->getKeyword("PORO");
-                const auto& ntg =  doubleGridProperties->getKeyword("NTG");
-
-                if (poro.containsNaN())
-                    throw std::logic_error("Do not have information for the PORV keyword - some defaulted values in PORO");
-                else {
-                    const auto& poroData = poro.getData();
-                    for (size_t globalIndex = 0; globalIndex < poro.getCartesianSize(); globalIndex++) {
-                        if (std::isnan(values[globalIndex])) {
-                            double cell_poro = poroData[globalIndex];
-                            double cell_ntg = ntg.iget(globalIndex);
-                            double cell_volume = eclipseGrid->getCellVolume(globalIndex);
-                            values[globalIndex] = cell_poro * cell_volume * cell_ntg;
-                        }
+            const auto& poro = doubleGridProperties->getKeyword("PORO");
+            const auto& ntg =  doubleGridProperties->getKeyword("NTG");
+
+            if (poro.containsNaN())
```

Comment:
This has slightly different semantics than the old solution. It is perfectly possible to fully initialize the `PORV` property from the deck with the `PORV` keyword, in that case the `PORO` keyword will be `NaN` all over the place - without any harm! So - the logical equivalent of the `std::find_if( ... )` - checking if there are `NaN` values in the `PORV` vector, must be performed before possibly raising an exception.

The normal path here will be that `values` vector is all `NaN` - I guess that information should influence the implementation. 

When this has been addressed I am happy to merge.